polygon-digital-carbon/src/RetirementHandler.ts

```diff
@@ -127,14 +127,16 @@ export function saveToucanPuroRetirementRequest(event: RetirementRequested): voi
   retire.consumptionCountryCode = event.params.params.consumptionCountryCode
   retire.consumptionPeriodStart = event.params.params.consumptionPeriodStart
   retire.consumptionPeriodEnd = event.params.params.consumptionPeriodEnd
-  retire.bridgeStatus = 'REQUESTED'
+  retire.asyncRetireStatus = AsyncRetireRequestStatus.REQUESTED
   retire.save()
 
-  let request = loadOrCreateToucanBridgeRequest(event.params.requestId)
+  let requestIdByteArray = Bytes.fromBigInt(event.params.requestId)
```

Comment:
Possible ID clashes if multiple registries start from 1 numerically? Puro is only passing the request ID while `getC3RetireRequestId` returns the combination of token + index.

Reply:
Combined asyncRetireRequest id generation into `createAsyncRetireRequestId` for both c3 and toucan, which takes the token address and index